x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/LifecycleExecutionState.java

```diff
@@ -310,8 +321,14 @@ public Builder setStepTime(Long stepTime) {
             return this;
         }
 
+        public Builder setIndexOriginationDate(Long originationDate) {
+            this.indexOriginationDate = originationDate;
+            return this;
+        }
+
         public LifecycleExecutionState build() {
-            return new LifecycleExecutionState(phase, action, step, failedStep, stepInfo, phaseDefinition, indexCreationDate,
+            return new LifecycleExecutionState(phase, action, step, failedStep, stepInfo, phaseDefinition,
+                indexOriginationDate != null && indexOriginationDate != -1L ? indexOriginationDate : indexCreationDate,
```

Comment:
See my above comment about not overwriting the `indexCreationDate`, if we only store one then a user can't go back to the "default" value for the date.